packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1526,7 +1526,9 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
     }
 
     final EdgeInsetsGeometry padding =
-        ButtonTheme.of(context).alignedDropdown ? _kAlignedButtonPadding : _kUnalignedButtonPadding;
+        ButtonTheme.of(context).alignedDropdown && widget._inputDecoration == null
```

Comment:
So DropdownButtonFormField customizes the InputDecoration which makes this padding irrelevant?

Reply:
Yes, this is based on https://github.com/flutter/flutter/pull/131183#discussion_r1396324819 and the corresponding PR, see https://github.com/flutter/flutter/pull/131183. I think it is the best option as removing the padding might break some DropdownButton usages.